scripts/zkai_native_attention_mlp_transcript_stable_comparison_gate.py

```
@@ -0,0 +1,870 @@
+    return output.getvalue()
+
+
+def require_output_path(path: pathlib.Path | None, suffix: str) -> pathlib.Path | None:
+    if path is None:
+        return None
+    candidate = ROOT / path if not path.is_absolute() else path
+    resolved_parent = candidate.parent.resolve()
+    resolved = resolved_parent / candidate.name
+    evidence_root = EVIDENCE_DIR.resolve()
+    if evidence_root not in resolved.parents:
+        raise TranscriptStableComparisonError("output path must stay under docs/engineering/evidence")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)